tests/components/test_mythicbeastsdns.py

```diff
@@ -0,0 +1,73 @@
+"""Test the Mythic Beasts DNS component."""
+import asyncio
+import logging
+import unittest
+
+from homeassistant.setup import async_setup_component
+from homeassistant.components import mythicbeastsdns
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TestMythicBeastsDNS(unittest.TestCase):
+    """Test class for Mythic Beasts DNS component."""
+
+    @unittest.mock.patch('mbddns.update')
```

Comment:
Standard unittest.mock.patch don't work in decorator mode for coroutines. You can use asynctest corresponding method as decorator or switch to context manager or use a pytest fixture.

Reply:
I'm having trouble getting the mock to apply outside of the class - if i use it in the class it appears to work ok, I've ripped out the class and replaced unittest.mock.patch with asynctest.mock.patch but it doesn't appear to actually apply to the calls.

```
@asynctest.mock.patch('mbddns.update')
async def mbddns_update_mock(domain, password, host, ttl=60,
                             session=None):
    """Mock out mythic beasts updater."""
```

Is how i've instantiated it, am i missing anything?

Reply:
The decorator is normally applied to a test to mock something before running the test. Looks like you're trying to apply the decorator to the function you want to use as side effect to the mocked function.